kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/PodPresetTest.java

```diff
@@ -0,0 +1,208 @@
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.mock;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+import io.fabric8.kubernetes.api.model.settings.PodPreset;
+import io.fabric8.kubernetes.api.model.settings.PodPresetBuilder;
+import io.fabric8.kubernetes.api.model.settings.PodPresetList;
+import io.fabric8.kubernetes.api.model.settings.PodPresetListBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.server.mock.KubernetesServer;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+public class PodPresetTest {
+  @Rule public KubernetesServer server = new KubernetesServer();
+
+  @Test
+  public void testList() {
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/test/podpresets").andReturn(200, new PodPresetListBuilder().build()).once();
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/ns1/podpresets").andReturn(200, new PodPresetListBuilder()
+      .addNewItem().and()
+      .addNewItem().and().build()).once();
+
+    KubernetesClient client = server.getClient();
+
+    PodPresetList podPresetList = client.settings().podPresets().list();
+    assertNotNull(podPresetList);
+    assertEquals(0, podPresetList.getItems().size());
+
+    podPresetList = client.settings().podPresets().inNamespace("ns1").list();
+    assertNotNull(podPresetList);
+    assertEquals(2, podPresetList.getItems().size());
+  }
+
+  @Test
+  public void testGet(){
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/test/podpresets/podpreset1").andReturn(200, new PodPresetBuilder().build()).once();
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/ns1/podpresets/podpreset2").andReturn(200, new PodPresetBuilder().build()).once();
+
+    KubernetesClient client = server.getClient();
+
+    PodPreset podpreset1 = client.settings().podPresets().withName("podpreset1").get();
+    assertNotNull(podpreset1);
+
+    PodPreset podpreset2 =  client.settings().podPresets().withName("podpreset2").get();
+    assertNull(podpreset2);
+
+    podpreset2 = client.settings().podPresets().inNamespace("ns1").withName("podpreset2").get();
+    assertNotNull(podpreset2);
+  }
+
+  @Test
+  public void testPodUpdate(){
+    KubernetesClient client = server.getClient();
+    try{
+      Pod pod = client.pods().load(getClass().getResourceAsStream("/test-podpreset.yml")).get();
+      assertEquals("pod-test", pod.getMetadata().getName());
+
+      server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/test/podpresets/podpreset-test").andReturn(200, new PodPresetBuilder()
```

Comment:
You should set server expectations first and then instantiate `KubernetesClient` from mocked server. This doesn't work.

Reply:
Oh alright, Understood